### PR TITLE
Run Jest e2e tests in parallel

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | https://github.com/babel/babel/pull/12202#discussion_r508357917
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/12233"></a>

### CODE REVIEW

File: scripts/integration-tests/e2e-jest.sh [hunk 1]
```diff
@@ -47,8 +47,6 @@ yarn build
 # The goals of this e2e test are:
 #   1) Check that the typescript compilation isn't completely broken
 #   2) Make sure that we don't accidentally break jest's usage of the Babel API
-CI=true yarn test-ci-partial packages
-CI=true yarn test-ci-partial e2e/__tests__/babel
-CI=true yarn test-ci-partial e2e/__tests__/transform
+CI=true yarn test-ci-partial:parallel --ci --max-workers $(nproc) packages e2e/__tests__/babel e2e/__tests__/transform
```
Comment (JLHwung) [inline scripts/integration-tests/e2e-jest.sh:50]: Nit: Can we delegate to Jest's default on max workers?

**Review by JLHwung — APPROVED** [review]

**Review by existentialism — APPROVED** [review]

**Review by kaicataldo — APPROVED** [review]

**Comment by nicolo-ribaudo** [comment]
Wait don't approve too much, the failing e2e test is jest :sweat_smile:

**Comment by JLHwung** [comment]
Yeah apparently circle-ci-bot requested changes so I didn't merge. 😂

**Comment by nicolo-ribaudo** [comment]
I tried SSH-ing in Circle CI to debug the failure, but I'm getting some `ENOMEM: not enough memory` errors (when running in parallel) that cause Jest to be in an invalid state end then other tests fail.

I'm going to close this PR for now, since it might just be that the tests need more memory than what Circle CI gives us.

**Pull request closed without merging**